modules/plotting/processors/dataframeexporter.cpp

```diff
@@ -84,67 +86,83 @@ void DataFrameExporter::exportNow() {
         LogWarn("File already exists: " << exportFile_);
         return;
     }
-
-    bool forceDoublePrecision = forceDoublePrecision_.get();
-    exportAsCSV(forceDoublePrecision);
+    exportAsCSV(separateVectorTypesIntoColumns_);
 }
 
-void DataFrameExporter::exportAsCSV(bool forceDoublePrecision) {
+void DataFrameExporter::exportAsCSV(bool separateVectorTypesIntoColumns) {
     std::ofstream file(exportFile_);
     auto dataFrame = dataFrame_.getData();
 
-    static std::string delimiter = ",";
-    static std::string lineterminator = "\n";
-    std::vector<std::string> componentNames = {"X", "Y", "Z", "W"};
+    const std::string delimiter = ", ";
+    const char lineterminator = '\n';
+    const std::array<char, 4> componentNames = {'X', 'Y', 'Z', 'W'};
 
     // headers
-    for (size_t j = 0; j < dataFrame->getNumberOfColumns(); j++) {
-        auto c = dataFrame->getColumn(j);
-        auto components = c->getBuffer()->getDataFormat()->getComponents();
-
-        if (components > 1 && forceDoublePrecision) {
+    auto oj = util::make_ostream_joiner(file, delimiter);
```

Comment:
won't a ostream_joiner add a delimiter after the last element as well?

Reply:
mine won't ;)